[docs] Update `request failed` error and troubleshooting guide (#5993)
DevExpress/testcafe: docs/articles/faq/README.md

When TestCafe does not receive a successful response from a server, it outputs the following error:

```text
Failed to load the page at https://www.example.com.
Increase the value of the "pageRequestTimeout" variable, enable the "retryTestPages" option, or use quarantine mode to perform additional attempts to execute this test.
```

> If your test is flaky, you can enable [quarantine mode](../documentation/guides/basic-guides/run-tests.md#quarantine-mode) to isolate it from the rest of the suite.

Reviewer: `flaky` Explain

Reviewer 2: https://docs.gitlab.com/ee/development/testing_guide/flaky_tests.html#whats-a-flaky-test @dirk-pieterse

Author: ```suggestion
If your test produces unstable results, you can enable [quarantine mode](../documentation/guides/basic-guides/run-tests.md#quarantine-mode) to isolate it from the rest of the suite.
```

> If your test produces unstable results, you can enable [quarantine mode](../documentation/guides/basic-guides/run-tests.md#quarantine-mode) to isolate it from the rest of the suite.

Author: ```suggestion
Use [quarantine mode](../documentation/guides/basic-guides/run-tests.md#quarantine-mode) to isolate *flaky* tests from the rest of the suite.
```

However, we recommend that you determine the cause of this issue and address it.

This error can occur in the following situations:

#### The Web server is not responding

Confirm the successful resolution of the website's domain name. Check if the web application is online and configured to receive incoming requests.

> If TestCafe does not receive the webpage within *two minutes*, it throws an exception.

Reviewer 2: I'd specify this fact, "if the TestCafe proxy..." yadda yadda 

Because it's hard for readers to distinguish b/w TestCafe the framework and TestCafe the test runner

> If the webpage is not served within *two minutes*, TestCafe throws an exception. Use one of the following options to increase the timeout value:

Reviewer: Clarify what you mean by `not served`

Author: ```suggestion
If TestCafe does not receive the webpage within *two minutes*, it throws an exception.
```

When TestCafe does not receive a successful response from a server, it outputs the following error:

```text
Failed to load a page at https://www.example.com.

> Increase the pageRequestTimeout, enable the retryTestPages option, or use quarantine mode to perform additional attempts to execute this test.

Reviewer 3: Add links to mentioned options.

> Increase the "pageRequestTimeout", enable the "retryTestPages" option, or use quarantine mode to perform additional attempts to execute this test.

Reviewer 3: or use "quarantineMode" option to

Author: ```suggestion
Increase the "pageRequestTimeout" or enable the "retryTestPages" option. 
You can use the "quarantineMode" option to perform additional attempts to execute this test.
```

> Increase the "pageRequestTimeout", enable the "retryTestPages" option, or use quarantine mode to perform additional attempts to execute this test.

Author: ```suggestion
Increase the "pageRequestTimeout", enable the "retryTestPages" option, or enter quarantine mode.
```

> Increase the "pageRequestTimeout", enable the "retryTestPages" option, or use quarantine mode to perform additional attempts to execute this test.

Reviewer 2: How about we meet halfway? what bothers you about the 'value' and the 'variable'?
```suggestion
Increase the value of the "pageRequestTimeout" variable, enable the "retryTestPages" option, or use quarantine mode to perform additional attempts to execute this test.
```

> Increase the "pageRequestTimeout", enable the "retryTestPages" option, or use quarantine mode to perform additional attempts to execute this test.

Author: It doesn't bother me at all (besides being somewhat redundant). `pageRequestTimeout` is self-explanatory, and doesn't call for an explanation (in my opinion). 

In general, I'm fine with your solution

```

> You can use [quarantine mode](../documentation/guides/basic-guides/run-tests.md#quarantine-mode) to complete the tests if this problem occurs infrequently.

Author: ```suggestion
If this error occurs inconsistently, you can enable [quarantine mode](../documentation/guides/basic-guides/run-tests.md#quarantine-mode) to isolate the problematic test from the rest of the suite.
```

> You can use [quarantine mode](../documentation/guides/basic-guides/run-tests.md#quarantine-mode) to complete the tests if this problem occurs infrequently.

Reviewer 2: An occurrence can not be inconsistent, but the rest of your suggestion is great 👍 
[flaky tests](https://docs.gitlab.com/ee/development/testing_guide/flaky_tests.html) are tests that sometimes fail, but sometimes succeed

how about
 ```suggestion
If your test is flaky, you can enable [quarantine mode](../documentation/guides/basic-guides/run-tests.md#quarantine-mode) to isolate it from the rest of the suite.
```

> You can use [quarantine mode](../documentation/guides/basic-guides/run-tests.md#quarantine-mode) to complete the tests if this problem occurs infrequently.

Author: This is great (for people who are acquainted with the term `flaky`). For others, it introduces an additional concept. 

But I guess the audience is knowledgeable enough to understand, so It's fine.

This error can occur in the following situations:

> #### The Web server is not responding or takes too long to respond

Reviewer 3: > or takes too long to respond

It's not necessary. The term `he Web server is not responding` means that the server is not responding for the allowed timeout.

Check if the Web and DNS servers are online and configured to accept requests to this URL.

> Before it throws an exception, TestCafe waits for *two minutes* for the webpage to be served. Use one of the following options to increase this timeout:

Author: ```suggestion
If the webpage is not served within *two minutes*, TestCafe throws an exception. Use one of the following options to increase the timeout value:
```

It is obvious that the server serves the webpage, hence the passive voice.

> Before it throws an exception, TestCafe waits for *two minutes* for the webpage to be served. Use one of the following options to increase this timeout:

Reviewer 2: Perfect

> * [`--page-request-timeout` CLI option](../documentation/reference/command-line-interface.md#--cache)

Reviewer 2: ```suggestion
* [`--page-request-timeout` (CLI)](../documentation/reference/command-line-interface.md#--cache)
```

Reviewer 2: either add the definite article (*the* 'timeout' option) or get rid of the word 'option'. you're repeating it either way

> * [`pageRequestTimeout` configuration file option](../documentation/reference/configuration-file.md#pagerequesttimeout)

Reviewer 2: ```suggestion
* [`pageRequestTimeout` (configuration file)](../documentation/reference/configuration-file.md#pagerequesttimeout)
```

> * [`pageRequestTimeout` option of the `runner.run` method](../documentation/reference/testcafe-api/runner/run.md)

Reviewer 2: ```suggestion
* [`pageRequestTimeout` (TestCafe Runner)](../documentation/reference/testcafe-api/runner/run.md)
```

> TestCafe can retry failed network request for webpages. Use one of the following options to enable retries:

Reviewer 2: I'd rather we avoid using the word 'retry' as a noun

```suggestion
TestCafe can automatically repeat unsuccessful HTTP requests. Use one of the following options to enable this behavior:
```

Author: It's not about all HTTP requests, just an initial HTTP Get request (when HTML itself is requested).

In my opinion, `to enable retries:` beats `to enable this behavior:`.

I suggest that we check this with the correctors.

Author: In my opinion, to `enable retries:` is better than `to enable this behavior:`

I suggest that we check this with the correctors.

Reviewer: How about: You can enable TestCafe to retry failed network requests for webpages. To do this, use one of the following options:

Or

You can use one of the following options enable TestCafe to retry failed network requests for webpages:

Author: ```suggestion
The following options enable TestCafe to retry failed network requests for webpages:
```

> * [`--retry-test-pages` CLI option](../documentation/reference/command-line-interface.md#--retry-test-pages)

Reviewer 2: ```suggestion
* [`--retry-test-pages` (CLI)](../documentation/reference/command-line-interface.md#--retry-test-pages)
```

> * [`retryTestPages` configuration file option](../documentation/reference/configuration-file.md#retrytestPages)

Reviewer 2: ```suggestion
* [`retryTestPages` (configuration file)](../documentation/reference/configuration-file.md#retrytestPages)
```

> * [`retryTestPages` option of the `runner.run` method](../documentation/reference/testcafe-api/runner/run.md)

Reviewer 2: ```suggestion
* [`retryTestPages` option (TestCafe Runner)](../documentation/reference/testcafe-api/runner/run.md)
```

#### Unstable or improperly configured network connection

* Check the network connection's settings.

> * Check your network equipment. If possible, establish a direct connection to the Internet/Web server.

Reviewer 2: what is a direct connection?
```suggestion
* Check your network equipment. If possible, establish a wired connection to the Internet/Web server.
```

Author: This is an odd recommendation. I'd rather:
```suggestion
* Check your network equipment and the connection to the Internet/Web server.
```

Reviewer 2: Nice!
